### DISCUSSION TITLE
Implementing WebSocket connection in Leptos SSR

### DISCUSSION THREAD

**Discussion by victorfds** [leptos-rs/leptos#2498] [Q&A] [answered]
I'm currently working on a project using Leptos **SSR** (Server-Side Rendering) and I'm interested in implementing WebSocket connections. I'm using the [start-axum-workspace](https://github.com/leptos-rs/start-axum-workspace) template as a base for my project.

Is there a recommended approach or best practice for implementing WebSocket connections in the Leptos SSR template? I'd appreciate any guidance or examples you can provide.

Thank you in advance for your help!

Best regards,

Victor Ferreira da Silva.

**Comment by gbj** [comment, marked as answer]
For the client side, you might like to explore [the `use_websocket` function from leptos-use](https://leptos-use.rs/network/use_websocket.html), as this should be pretty smooth to use with the rest of the framework! For the server end of the websocket, I don't think there are any Leptos particulars; you can use whichever Axum-compatible websocket library you'd like. (I am not very familiar with them or I would offer a recommendation.)

**Reply by victorfds** [reply]
Thank you very much, @gbj , I will proceed with your recommendation.
